scraper/scraper.ts

```diff
@@ -1,28 +1,974 @@
+  const censusDates = formatDates(course.censusDates)
+  const currentYear = new Date().getFullYear()
+
+  // Add the current year to the date and convert it to a date
+  // object so it can be easily parsed
+  const refCensusDates = formatDates(
+    reference.map(element => {
+      return element.census + '/' + currentYear
+    })
+  )
+
+  const termList: Term[] = []
+  for (const censusDate of censusDates) {
+    // minimum difference with the reference census date means the course is in that term
+    let min = -1
+    let index = 0
+    let term: Term
+    for (const refDate of refCensusDates) {
+      // Calculate the difference (in days) between the
+      // census date for the term and the reference census date
+      const diff = Math.abs(
+        Math.round(
+          (censusDate.getMonth() - refDate.getMonth()) * 30 +
+            censusDate.getDate() -
+            refDate.getDate()
+        )
+      )
+      if (diff < min || min === -1) {
+        min = diff
+        term = reference[index].term
+      }
+      index++
+    }
+    termList.push(term)
+  }
+  return termList
+}
+
+/**
+ * Finds the Term for a class. Term is defined in interfaces.ts
+ * @param cls: Class to find the term for
+ * @param reference: Refernce dates to find the term.
+ * Format of each element: { term: Term, dates: { start: number[], length: number[] } }
+ * start is an array of possible start dates and length is
+ */
+const classTermFinder = (
+  cls: Class,
+  reference = [
+    {
+      term: Term.Summer,
+      dates: [{ start: 11, length: 3 }, { start: 12, length: 2 }],
+    },
+    {
+      term: Term.T1,
+      dates: [
+        { start: 1, length: 2 },
+        { start: 1, length: 3 },
+        { start: 1, length: 4 },
+        { start: 2, length: 1 },
+        { start: 2, length: 3 },
+        { start: 3, length: 1 },
+      ],
+    },
+    {
+      term: Term.T2,
+      dates: [
+        { start: 3, length: 2 },
+        { start: 4, length: 2 },
+        { start: 4, length: 3 },
+        { start: 5, length: 1 },
+        { start: 5, length: 3 },
+        { start: 6, length: 1 },
+        { start: 6, length: 3 },
+        { start: 7, length: 1 },
+        { start: 7, length: 2 },
+        { start: 8, length: 1 },
+      ],
+    },
+    {
+      term: Term.T3,
+      dates: [
+        { start: 8, length: 2 },
+        { start: 8, length: 3 },
+        { start: 9, length: 1 },
+        { start: 9, length: 2 },
+        { start: 9, length: 3 },
+        { start: 10, length: 1 },
+        { start: 10, length: 2 },
+      ],
+    },
+    {
+      term: Term.S1,
+      dates: [{ start: 2, length: 4 }],
+    },
+    { term: Term.S2, dates: [{ start: 7, length: 4 }] },
+  ]
+): Term => {
+  // Error check
+  if (!(cls && cls.termDates)) {
+    throw new Error('no start and end dates for class: ' + cls)
+  }
+
+  const [start, end] = formatDates(
+    [cls.termDates.start, cls.termDates.end].map(date =>
+      reverseDayAndMonth(date, '/')
```

Comment:
Can change the formatDates call outside to be moved inside, refactor function to take in one string and return one date rather than operating on lists of strings and returning list of dates
```suggestion
      formatDate(reverseDayAndMonth(date, '/'))
```

Reply:
Think I prefer it this way